### ISSUE TITLE
When deploying from Parent -> Child, UDA files are not being extracted to the backoffice.

### ISSUE THREAD

**Issue by thejokerd3** [umbraco/Umbraco.Cloud.Issues#492]
### Issue description

When pushing metadata updates from a baseline to a child project, then the changes are not reflected in the backoffice.
Meaning that you need to run a manual extraction of UDA files in the data folder to get the changes reflected in the backoffice.

Multiple tests has been done with this.

This is the baseline alias: afac

This is the child alias: nordic-firework

This is the deployment Log of the Child project:

```
11:02:36: Updating submodules.
11:02:37: Preparing deployment for commit id '53a8aaf205'.
11:02:37: Running custom deployment command...
11:02:37: Running deployment command...
11:02:37: Command: C:\home\SiteExtensions\Umbraco.Cloud.Deployment.SiteExtension.Artifacts\deploy.cmd
11:02:37: Umbraco Cloud deployment
11:02:37: Handling Basic Web Site deployment.
11:02:39: KuduSync.NET from: 'C:\home\site\repository' to: 'C:\home\site\deployments\___deployTemp'
11:02:39: Copying file: 'Latch.Web.live.xdt.config'
11:02:39: Copying file: 'Web.development.xdt.config'
11:02:39: Copying file: 'Web.live.xdt.config'
11:02:39: Copying file: 'Web.staging.xdt.config'
11:02:39: Copying file: 'Config\serilog.user.config'
11:02:39: Copying file: 'Config\imageprocessor\BlobStorage.cache.development.xdt.config'
11:02:39: Copying file: 'Config\imageprocessor\BlobStorage.cache.live.xdt.config'
11:02:39: Copying file: 'Config\imageprocessor\BlobStorage.cache.staging.xdt.config'
11:02:39: Copying file: 'Config\imageprocessor\BlobStorage.security.development.xdt.config'
11:02:39: Copying file: 'Config\imageprocessor\BlobStorage.security.live.xdt.config'
11:02:39: Copying file: 'Config\imageprocessor\BlobStorage.security.staging.xdt.config'
11:02:39: Copying file: 'Config\imageprocessor\cache.config'
11:02:39: Copying file: 'Config\imageprocessor\cache.development.xdt.config'
11:02:39: Copying file: 'Config\imageprocessor\cache.live.xdt.config'
11:02:39: Copying file: 'Config\imageprocessor\cache.staging.xdt.config'
11:02:42: Copying file: 'Config\imageprocessor\security.config'
11:02:42: Copying file: 'Config\imageprocessor\security.development.xdt.config'
11:02:42: Copying file: 'Config\imageprocessor\security.live.xdt.config'
11:02:42: Copying file: 'Config\imageprocessor\security.staging.xdt.config'
11:02:42: Copying file: 'data\revision\document-type__0c3d7343538f4cef9d92d18875cadb77.uda'
11:02:42: Copying file: 'Media\BlobStorage.web.development.xdt.config'
11:02:42: Copying file: 'Media\BlobStorage.web.live.xdt.config'
11:02:42: Copying file: 'Media\BlobStorage.web.staging.xdt.config'
11:02:42: Copying file: 'Web.config'
11:02:42: Copying file: 'Media\Web.config'
11:02:43: Preparing deployment
11:02:43: Checking file Examine.dll for version 0.1.80 to verify transforms.
11:02:43: Current version is: 1.1.0
11:02:43: Checking file ImageProcessor.Web.dll for version 4.7.1 to verify transforms.
11:02:43: Current version is: 4.12.1.00216
11:02:43: Checking file Umbraco.Core.dll for version 7.5.11 to verify transforms.
11:02:43: Current version is: 8.13.1
11:02:43: Checking file Umbraco.Core.dll for version 7.3.0 to verify transforms.
11:02:43: Current version is: 8.13.1
11:02:43: Site is not using log4net - Skipping version check
11:02:43: Checking file Umbraco.Core.dll for version 8.0.0 to verify transforms.
11:02:43: Current version is: 8.13.1
11:02:43: Handling Umbraco Cloud default config transform
11:02:43: Handling custom config transforms for live
11:02:43: Transforming Web.config using Latch.Web.live.xdt.config
11:02:44: Transforming Web.config using Web.live.xdt.config
11:02:44: Transforming cache.config using BlobStorage.cache.live.xdt.config
11:02:46: Transforming security.config using BlobStorage.security.live.xdt.config
11:02:46: Transforming cache.config using cache.live.xdt.config
11:02:46: Transforming security.config using security.live.xdt.config
11:02:46: Transforming web.config using BlobStorage.web.live.xdt.config
11:02:47: Handling Umbraco Cloud version specific config transform
11:02:48: Handling Umbraco Cloud specific config transform
11:02:48: Handling live config transform
11:02:51: Finished successfully.
11:02:51: Deploying changes to wwwroot
11:02:58: KuduSync.NET from: 'C:\home\site\deployments\___deployTemp' to: 'C:\home\site\wwwroot'
11:02:58: Copying file: 'data\revision\document-type__0c3d7343538f4cef9d92d18875cadb77.uda'
11:02:58: Checking for Umbraco Deploy Uda files
11:02:58: Current environment: live
11:02:58: Checking for changes in Umbraco Deploy metadata files in 'first environment (live only)'
11:02:58: Checking for changes between 0000000000000000000000000000000000000000 and 26027eae38470cc4ed69d880c320fd9744b8bcc6
11:02:59: git : fatal: bad object 0000000000000000000000000000000000000000
11:02:59: =
11:02:59: At C:\home\SiteExtensions\Umbraco.Cloud.Deployment.SiteExtension.Artifacts\umbr
11:02:59: No changes in metadata detected - no need to run an Umbraco Deploy extraction
11:02:59: acodeploy\EnsureUmbracoDeployExtraction.ps1:75 char:22
11:02:59: Finished successfully.
11:02:59: + ... cmdOutput = git diff --name-only $fromCommit $toCommit $repositoryPat ...
11:02:59: + ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
11:02:59: + CategoryInfo : NotSpecified: (fatal: bad obje...000000000000000
11:02:59: :String) [], RemoteException
11:02:59: + FullyQualifiedErrorId : NativeCommandError
11:02:59:
11:02:59: Running post deployment command(s)...
11:02:59: Triggering recycle (preview mode disabled).
11:02:59: Deployment successful.
```

**Comment by thejokerd3** [comment]
This is happening on the new platform.

Due notice - that the metadata (UDA files) are being pushed from Parent -> child respectively to the repository folder, and extracted to wwwroot. But the extraction from wwwroot is not happening.

**Comment by sitereactor** [comment]
This is fixed by adding a fallback option to the script that does the comparison, which checks for Umbraco Deploy metadata.
Internal reference #15025 and PR1390

**Comment by sitereactor** [comment]
Umbraco.Cloud.Deployment.SiteExtension.Artifacts.Core for Umbraco 9 projects is available with version 1.0.4
and Umbraco.Cloud.Deployment.SiteExtension.Artifacts for Umbraco 7+8 is available with version 1.1.11

**Comment by thejokerd3** [comment]
It unfortunately still doesn't work between these two projects:
Parent: afac
Child: nordic-firework

Siteextension was updated on BOTH projects, and on all environments respectively.

But it seems that the child project can't register changes in metadata

`14:13:44: Updating submodules.
14:13:44: Preparing deployment for commit id 'e3eea93c32'.
14:13:44: Running custom deployment command...
14:13:44: Running deployment command...
14:13:44: Command: C:\home\SiteExtensions\Umbraco.Cloud.Deployment.SiteExtension.Artifacts\deploy.cmd
14:13:44: Umbraco Cloud deployment
14:13:44: Handling Basic Web Site deployment.
14:13:46: KuduSync.NET from: 'C:\home\site\repository' to: 'C:\home\site\deployments\___deployTemp'
14:13:46: Copying file: 'Latch.Web.live.xdt.config'
14:13:46: Copying file: 'Web.development.xdt.config'
14:13:46: Copying file: 'Web.live.xdt.config'
14:13:46: Copying file: 'Web.staging.xdt.config'
14:13:46: Copying file: 'Config\serilog.user.config'
14:13:46: Copying file: 'Config\imageprocessor\BlobStorage.cache.development.xdt.config'
14:13:46: Copying file: 'Config\imageprocessor\BlobStorage.cache.live.xdt.config'
14:13:46: Copying file: 'Config\imageprocessor\BlobStorage.cache.staging.xdt.config'
14:13:46: Copying file: 'Config\imageprocessor\BlobStorage.security.development.xdt.config'
14:13:46: Copying file: 'Config\imageprocessor\BlobStorage.security.live.xdt.config'
14:13:46: Copying file: 'Config\imageprocessor\BlobStorage.security.staging.xdt.config'
14:13:46: Copying file: 'Config\imageprocessor\cache.config'
14:13:46: Copying file: 'Config\imageprocessor\cache.development.xdt.config'
14:13:46: Copying file: 'Config\imageprocessor\cache.live.xdt.config'
14:13:46: Copying file: 'Config\imageprocessor\cache.staging.xdt.config'
14:13:49: Copying file: 'Config\imageprocessor\security.config'
14:13:49: Copying file: 'Config\imageprocessor\security.development.xdt.config'
14:13:49: Copying file: 'Config\imageprocessor\security.live.xdt.config'
14:13:49: Copying file: 'Config\imageprocessor\security.staging.xdt.config'
14:13:49: Copying file: 'data\revision\document-type__491f7ffe4bdd4ed08221e083b39962a4.uda'
14:13:49: Copying file: 'Media\BlobStorage.web.development.xdt.config'
14:13:49: Copying file: 'Media\BlobStorage.web.live.xdt.config'
14:13:49: Copying file: 'Media\BlobStorage.web.staging.xdt.config'
14:13:49: Copying file: 'Web.config'
14:13:49: Copying file: 'Media\Web.config'
14:13:49: Preparing deployment
14:13:49: Checking file Examine.dll for version 0.1.80 to verify transforms.
14:13:49: Current version is: 1.1.0
14:13:49: Checking file ImageProcessor.Web.dll for version 4.7.1 to verify transforms.
14:13:49: Current version is: 4.12.1.00216
14:13:49: Checking file Umbraco.Core.dll for version 7.5.11 to verify transforms.
14:13:49: Current version is: 8.13.1
14:13:50: Checking file Umbraco.Core.dll for version 7.3.0 to verify transforms.
14:13:50: Current version is: 8.13.1
14:13:50: Site is not using log4net - Skipping version check
14:13:50: Checking file Umbraco.Core.dll for version 8.0.0 to verify transforms.
14:13:50: Current version is: 8.13.1
14:13:50: Handling Umbraco Cloud default config transform
14:13:50: Handling custom config transforms for live
14:13:50: Transforming Web.config using Latch.Web.live.xdt.config
14:13:50: Transforming Web.config using Web.live.xdt.config
14:13:50: Transforming cache.config using BlobStorage.cache.live.xdt.config
14:13:50: Transforming security.config using BlobStorage.security.live.xdt.config
14:13:51: Transforming cache.config using cache.live.xdt.config
14:13:51: Transforming security.config using security.live.xdt.config
14:13:51: Transforming web.config using BlobStorage.web.live.xdt.config
14:13:52: Handling Umbraco Cloud version specific config transform
14:13:53: Handling Umbraco Cloud specific config transform
14:13:53: Handling live config transform
14:13:56: Finished successfully.
14:13:56: Deploying changes to wwwroot
14:14:01: KuduSync.NET from: 'C:\home\site\deployments\___deployTemp' to: 'C:\home\site\wwwroot'
14:14:01: Copying file: 'data\revision\document-type__491f7ffe4bdd4ed08221e083b39962a4.uda'
14:14:01: Checking for Umbraco Deploy Uda files
14:14:02: Current environment: live
14:14:02: Checking for changes in Umbraco Deploy metadata files in 'first environment (live only)'
14:14:02: Checking for changes between 6e290efd607a219a494b203faeee5a60c452986f
14:14:02: and 26027eae38470cc4ed69d880c320fd9744b8bcc6
14:14:02: git : fatal: ambiguous argument '6e290efd607a219a494b203faeee5a60c452986f?
14:14:02: =
14:14:02: At C:\home\SiteExtensions\Umbraco.Cloud.Deployment.SiteExtension.Artifacts\umbr
14:14:02: No changes in metadata detected - no need to run an Umbraco Deploy extraction
14:14:02: acodeploy\EnsureUmbracoDeployExtraction.ps1:80 char:22
14:14:02: + ... cmdOutput = git diff --name-only $fromCommit $toCommit $repositoryPat ...
14:14:02: + ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
14:14:02: Finished successfully.
14:14:02: + CategoryInfo : NotSpecified: (fatal: ambiguou...ee5a60c452986f?
14:14:02: :String) [], RemoteException
14:14:02: + FullyQualifiedErrorId : NativeCommandError
14:14:02:
14:14:02: ': unknown revision or path not in the working tree.
14:14:02: Use '--' to separate paths from revisions, like this:
14:14:02: 'git <command> [<revision>...] -- [<file>...]'
14:14:02: Running post deployment command(s)...
14:14:02: Triggering recycle (preview mode disabled).
14:14:02: Deployment successful.`

**Comment by sitereactor** [comment]
Formatting update pushed and available in ...: Umbraco.Cloud.Deployment.SiteExtension.Artifacts.Core for Umbraco 9 projects is available with version 1.0.5
and Umbraco.Cloud.Deployment.SiteExtension.Artifacts for Umbraco 7+8 is available with version 1.1.12
